File: scripts/check_schema.py
#!/usr/bin/env python3

import os
from dotenv import load_dotenv
from supabase import create_client

# Load environment variables
load_dotenv(dotenv_path="../.env")

# Get credentials
supabase_url = os.environ.get("NEXT_PUBLIC_SUPABASE_URL")
supabase_key = os.environ.get("SUPABASE_SERVICE_ROLE_KEY")

print("🔍 CHECKING DATABASE SCHEMA")
print("=" * 50)

try:
    # Create client
    supabase = create_client(supabase_url, supabase_key)
    
    # Check if profiles table exists and what columns it has
    print("📋 Checking profiles table structure...")
    
    # Query information schema to see table structure
    result = supabase.rpc('exec_sql', {
        'sql': """
        SELECT column_name, data_type, is_nullable 
        FROM information_schema.columns 
        WHERE table_name = 'profiles' 
        ORDER BY ordinal_position;
        """
    }).execute()
    
    if result.data:
        print("✅ Profiles table columns:")
        for row in result.data:
            print(f"   - {row['column_name']}: {row['data_type']} ({'NULL' if row['is_nullable'] == 'YES' else 'NOT NULL'})")
    else:
        print("❌ Could not get table structure or table doesn't exist")
    
    # Check for climate_focus column specifically
    climate_focus_check = supabase.rpc('exec_sql', {
        'sql': """
        SELECT EXISTS (
            SELECT 1 FROM information_schema.columns 
            WHERE table_name = 'profiles' AND column_name = 'climate_focus'
        ) as climate_focus_exists;
        """
    }).execute()
    
    if climate_focus_check.data:
        exists = climate_focus_check.data[0]['climate_focus_exists']
        print(f"\n🎯 climate_focus column exists: {'✅ YES' if exists else '❌ NO'}")
        
        if not exists:
            print("\n📝 MIGRATION NEEDED:")
            print("   Run the updated migration file in Supabase SQL Editor:")
            print("   supabase/migrations/20240708000000_climate_economy_ecosystem.sql")
    
except Exception as e:
    print(f"❌ Error checking schema: {e}")
    print("\n💡 This might be because the exec_sql function doesn't exist.")
    print("   Please run the migration file manually in Supabase SQL Editor.") 

File: scripts/check-schema.js
const { createClient } = require('@supabase/supabase-js');
require('dotenv').config();

const supabase = createClient(
  process.env.NEXT_PUBLIC_SUPABASE_URL, 
  process.env.SUPABASE_SERVICE_ROLE_KEY
);

async function checkSchema() {
  try {
    console.log('🔍 Checking database schema...\n');
    
    // Check job_seeker_profiles
    try {
      const { data: jobSeekerData } = await supabase
        .from('job_seeker_profiles')
        .select('*')
        .limit(1);
      console.log('📋 job_seeker_profiles columns:');
      console.log(Object.keys(jobSeekerData?.[0] || {}));
      console.log('Sample data:', jobSeekerData?.[0] || 'No data');
    } catch (error) {
      console.log('❌ job_seeker_profiles error:', error.message);
    }
    
    console.log('\n' + '='.repeat(50) + '\n');
    
    // Check partner_profiles  
    try {
      const { data: partnerData } = await supabase
        .from('partner_profiles')
        .select('*')
        .limit(1);
      console.log('📋 partner_profiles columns:');
      console.log(Object.keys(partnerData?.[0] || {}));
      console.log('Sample data:', partnerData?.[0] || 'No data');
    } catch (error) {
      console.log('❌ partner_profiles error:', error.message);
    }
    
    console.log('\n' + '='.repeat(50) + '\n');
    
    // Check admin_profiles
    try {
      const { data: adminData } = await supabase
        .from('admin_profiles')
        .select('*')
        .limit(1);
      console.log('📋 admin_profiles columns:');
      console.log(Object.keys(adminData?.[0] || {}));
      console.log('Sample data:', adminData?.[0] || 'No data');
    } catch (error) {
      console.log('❌ admin_profiles error:', error.message);
    }
    
  } catch (error) {
    console.error('❌ Schema check failed:', error.message);
  }
}

checkSchema(); 